sdk/core/core-lro/review/core-lro.api.md

```diff
@@ -10,11 +10,26 @@ import { AbortSignalLike } from '@azure/abort-controller';
 export type CancelOnProgress = () => void;
 
 // @public
-export interface LongRunningOperation<T> {
-    requestMethod: string;
-    requestPath: string;
-    sendInitialRequest: () => Promise<LroResponse<T>>;
-    sendPollRequest: (path: string) => Promise<LroResponse<T>>;
+export function createPoller<TResult, TState extends OperationState<TResult>>(lro: LongRunningOperation, options?: CreatePollerOptions<TResult, TState>): Promise<SimplePollerLike<TState, TResult>>;
+
+// @public
+export interface CreatePollerOptions<TResult, TState> {
+    intervalInMs?: number;
+    processResult?: (result: unknown, state: TState) => TResult;
+    resourceLocationConfig?: LroResourceLocationConfig;
+    restoreFrom?: string;
+    updateState?: (state: TState, lastResponse: RawResponse) => void;
+    withPollingUrl?: (pollingUrl: string) => void;
+}
```

Comment:
This abstraction feels a bit too leaky. It exposes a lot of HTTP-specific and Azure-specific logic in the API.

I think there's room for a two-layered abstraction. Something like `createAzurePoller` which does what you have written here and is specific to the Azure HTTP LRO pattern, and an even smaller `createPoller` that encapsulates only the logic around managing updates to the poller state.

It might even make sense to do a `core-poller` package where the simplest form of the poller can live and then import/use that in core-lro, where the Azure convention bits can live.

Reply:
Could you elaborate on what scenario you're trying to accommodate here?

Reply:
@mpodwysocki could you share some of what you showed me you are trying to accomplish in NH with the polling API?

I'm also not 100% sure how I'd take the existing logic in FR and encode it in `createPoller`. What we have there is so simple, but I'm not sure how it will map into LongRunningOperation/CreatePollerOptions. In FR there's just a function to init the state, a function to poll, and some helpers for serialization/deserialization.

Reply:
> I'm also not 100% sure how I'd take the existing logic in FR and encode it in createPoller. What we have there is so simple, but I'm not sure how it will map into LongRunningOperation/CreatePollerOptions. In FR there's just a function to init the state, a function to poll, and some helpers for serialization/deserialization.

It is pretty much simpler, mainly, you will need to specify:
- sendInitRequest: to send the operation initiation request
- sendPollRequest: to send the polling GET request given the operation location URL as input
- processResult: to transform the result in the final response to a convenience-layer result
- updateState: to update the poller's state

You no longer need to worry about how the poller's state is being instantiated or updated to perform the polling logic. See https://github.com/Azure/azure-sdk-for-js/pull/22852 for an example in using it in @azure/ai-language-text.

Reply:
I created a protocol-agnostic poller and used it to implement the http one in [5baa826](https://github.com/Azure/azure-sdk-for-js/pull/22753/commits/5baa826678bbc2e36b4f7052d9e7775294eab9df). How does that look?